gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -217,41 +238,51 @@ private void validate(GoogleCloudStorageItemInfo itemInfo) throws IOException {
   private GoogleCloudStorageItemInfo getObjectMetadata(StorageResourceId resourceId, Storage gcs)
       throws IOException {
     StorageObject object;
-    long startTime = System.currentTimeMillis();
-    try {
-      // TODO(b/190617054) : Migrate to gRPC requests for metadata when available
-      // Request only fields that are used for metadata initialization
-      Get metadataRequest = getMetadataRequest(gcs, resourceId).setFields(METADATA_FIELDS);
-      object =
-          ResilientOperation.retry(
-              metadataRequest::execute,
-              backOffFactory.newBackOff(),
-              RetryDeterminer.SOCKET_ERRORS,
```

Comment:
Good catch, it was unintended. I re-wrote based on what the metadata API can throw. 
Considering that the api can through IOException, i think it could be useful to retry more than SOCKET_ERRORS, For now, i think we could revert.

Reply:
I still see retries for all IOException